Cesium.CodeGen/Ir/Expressions/Constants/IntegerConstant.cs

```diff
@@ -44,4 +44,42 @@ public void EmitTo(IEmitScope scope)
     public IType GetConstantType(IDeclarationScope scope) => scope.CTypeSystem.Int;
 
     public override string ToString() => $"integer: {Value}";
+
+    private static bool TryParse(string text, out int value)
+    {
+        if (text.StartsWith("0x"))
+        {
+            if (int.TryParse(text[2..], System.Globalization.NumberStyles.HexNumber, null, out value))
+            {
+                return true;
+            }
+
+            throw new CompilationException($"Invalid hex number {text}");
+        }
+
+        if (text.StartsWith("0"))
+        {
+            value = 0;
+            for (var i = 0; i < text.Length; i++)
+            {
+                if (text[i] >= '0' && text[i] <= '7')
+                {
+                    value = 8 * value + (text[i] - '0');
+                }
+                else
+                {
+                    throw new CompilationException($"Invalid octal number {text}");
+                }
+            }
+
+            return true;
+        }
+
+        if (int.TryParse(text, out value))
+        {
+            return true;
+        }
```

Comment:
I'll change this to pass the invariant culture to all the `TryParse` calls here.